tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/client/IngressTimeTest.java

```diff
@@ -0,0 +1,145 @@
+   }
+
+   @Test
+   public void testSendCoreReceiveCore() throws Throwable {
+      internalSendReceive(Protocol.CORE, Protocol.CORE);
+   }
+
+   @Test
+   public void testSendAMQPReceiveCore() throws Throwable {
+      internalSendReceive(Protocol.AMQP, Protocol.CORE);
+   }
+
+   @Test
+   public void testSendOpenWireReceiveCore() throws Throwable {
+      internalSendReceive(Protocol.OPENWIRE, Protocol.CORE);
+   }
+
+   @Test
+   public void testSendCoreReceiveOpenwire() throws Throwable {
+      internalSendReceive(Protocol.CORE, Protocol.OPENWIRE);
+   }
+
+   @Test
+   public void testSendAMQPReceiveOpenWire() throws Throwable {
+      internalSendReceive(Protocol.AMQP, Protocol.OPENWIRE);
+   }
+
+   @Test
+   public void testSendOpenWireReceiveOpenWire() throws Throwable {
+      internalSendReceive(Protocol.OPENWIRE, Protocol.OPENWIRE);
+   }
+
+   /**
+    *  AMQP receiver tests are in org.apache.activemq.artemis.tests.integration.amqp.AmqpIngressTimeTest
+    */
+
+   private void internalSendReceive(Protocol protocolSender, Protocol protocolConsumer) throws Throwable {
+      ConnectionFactory factorySend = createFactory(protocolSender);
+      ConnectionFactory factoryConsume = protocolConsumer == protocolSender ? factorySend : createFactory(protocolConsumer);
+
+      long beforeSend, afterSend;
+      try (Connection connection = factorySend.createConnection()) {
+         try (Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)) {
+            javax.jms.Queue queue = session.createQueue(QUEUE.toString());
+            try (MessageProducer producer = session.createProducer(queue)) {
+               producer.setDeliveryMode(DeliveryMode.PERSISTENT);
+               TextMessage msg = session.createTextMessage("hello");
+               beforeSend = System.currentTimeMillis();
+               producer.send(msg);
+               afterSend = System.currentTimeMillis();
+            }
+         }
+      }
+
+      server.stop();
+      server.start();
+      assertTrue(server.waitForActivation(3, TimeUnit.SECONDS));
+
+      try (Connection connection = factoryConsume.createConnection()) {
+         connection.start();
+         try (Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)) {
+            javax.jms.Queue queue = session.createQueue(QUEUE.toString());
+            try (MessageConsumer consumer = session.createConsumer(queue)) {
+               TextMessage message = (TextMessage) consumer.receive(1000);
+               Assert.assertNotNull(message);
+               Object ingressTime = message.getObjectProperty(Message.HDR_INGRESS_TIME.toString());
+               assertNotNull(ingressTime);
+               assertTrue(ingressTime instanceof Long);
+               long ingress = (Long) ingressTime;
+               assertTrue("Ingress time should be set in expected range",beforeSend <= ingress && afterSend >= ingress);
```

Comment:
Same point as last time about the unhelpful failure this will produce

Reply:
Done.